### PR TITLE
feat: update to latest proofs dependencies

### PR DESCRIPTION
**Pull request by cryptonemo**
fix: update proofs code to accomodate for the v6.x re-factors

### CODE REVIEW

**Review by dignifiedquire — CHANGES_REQUESTED** [review]
we should proxy the new `gpu2` flag through

File: Cargo.toml [hunk 1]
```diff
@@ -10,13 +10,17 @@ readme = "README.md"
...
+filecoin-hashers = { version = "~1.1", default-features = false, features = ["poseidon", "sha256"] }
+fr32 = { version = "~0.2", default-features = false }
+storage-proofs-core = { version = "~6.1.0", default-features = false }
 
 [features]
 default = ["pairing"]
 pairing = ["filecoin-proofs-v1/pairing", "bellperson/pairing"]
 blst = ["filecoin-proofs-v1/blst", "bellperson/blst"]
+gpu = ["filecoin-proofs-v1/gpu", "filecoin-hashers/gpu", "storage-proofs-core/gpu", "fr32/gpu"]
+gpu2 = ["filecoin-proofs-v1/gpu2", "filecoin-hashers/gpu2", "storage-proofs-core/gpu2", "fr32/gpu"]
```
[the remark this answers is not in the document]
Comment (cryptonemo) [reply Cargo.toml:26]: I don't think bellperson has gpu2

[the remark this answers is not in the document]
Comment (dignifiedquire) [reply Cargo.toml]: I know, but if because `gpu` is optional for all except for bellperson now

[the remark this answers is not in the document]
Comment (cryptonemo) [reply Cargo.toml]: I'm not following.  Is there something specific missing that hasn't been pushed yet?

**Review by dignifiedquire — APPROVED** [review]
now it looks like I imagined :)

**Pull request merged**